pkg/odo/genericclioptions/runnable.go

```diff
@@ -160,27 +160,7 @@ func checkConflictingFlags(cmd *cobra.Command, args []string) error {
 	project := stringFlagLookup(cmd, "project")
 	context := stringFlagLookup(cmd, "context")
 	component := stringFlagLookup(cmd, "component")
-	all, _ := strconv.ParseBool(stringFlagLookup(cmd, "all"))
-	// TODO: Move this to a method under DeleteOptions, similar to CreateOptions.checkConflictingFlags
-	if cmd.Name() == "v2delete" {
-		if cmd.HasParent() {
-			if cmd.Parent().Name() == "odo" || cmd.Parent().Name() == "component" {
-				var componentName string
-				if len(args) > 0 {
-					componentName = args[0]
-				}
-				if (context != "") && (project != "" || componentName != "") {
-					return fmt.Errorf("cannot provide --project or component name when --context is provided")
-				}
-				if project != "" && componentName == "" && app == "" {
-					return fmt.Errorf("cannot provide --project without --app and component name")
-				}
-				if all && ((componentName != "" && app != "" && project != "") || (componentName != "")) {
-					return fmt.Errorf("cannot provide --all when component name, --app and --project are provided")
-				}
-			}
-		}
-	}
+	// TODO: add validations for the delete command
```

Comment:
Will this be added as a part of this PR or a separate issue/PR?

Reply:
Separate PR/issue.

Reply:
Or perhaps this is not required. I have removed the TODO for now. We can add more validations as and when they're encountered.